miner/scroll_worker.go

```diff
@@ -128,29 +161,33 @@ type worker struct {
 	// External functions
 	isLocalBlock func(block *types.Block) bool // Function used to determine whether the specified block is mined by local miner.
 
-	circuitCapacityChecker *ccc.Checker
-	prioritizedTx          *prioritizedTransaction
+	prioritizedTx *prioritizedTransaction
+
+	asyncChecker *ccc.AsyncChecker
 
 	// Test hooks
 	beforeTxHook func() // Method to call before processing a transaction.
+
+	errCountdown int
+	skipTxHash   common.Hash
 }
 
 func newWorker(config *Config, chainConfig *params.ChainConfig, engine consensus.Engine, eth Backend, mux *event.TypeMux, isLocalBlock func(*types.Block) bool, init bool) *worker {
 	worker := &worker{
-		config:                 config,
-		chainConfig:            chainConfig,
-		engine:                 engine,
-		eth:                    eth,
-		mux:                    mux,
-		chain:                  eth.BlockChain(),
-		isLocalBlock:           isLocalBlock,
-		txsCh:                  make(chan core.NewTxsEvent, txChanSize),
-		chainHeadCh:            make(chan core.ChainHeadEvent, chainHeadChanSize),
-		exitCh:                 make(chan struct{}),
-		startCh:                make(chan struct{}, 1),
-		circuitCapacityChecker: ccc.NewChecker(true),
-	}
-	log.Info("created new worker", "CircuitCapacityChecker ID", worker.circuitCapacityChecker.ID)
+		config:       config,
+		chainConfig:  chainConfig,
+		engine:       engine,
+		eth:          eth,
+		mux:          mux,
+		chain:        eth.BlockChain(),
+		isLocalBlock: isLocalBlock,
+		txsCh:        make(chan core.NewTxsEvent, txChanSize),
+		chainHeadCh:  make(chan core.ChainHeadEvent, chainHeadChanSize),
+		exitCh:       make(chan struct{}),
+		startCh:      make(chan struct{}, 1),
+		reorgCh:      make(chan reorgTrigger, 2*(config.CCCMaxWorkers+1)),
```

Comment:
Would `config.CCCMaxWorkers+1` be sufficient for this queue?

Reply:
It might be, but if I am not mistaken it needs to be able to hold atleast 2 reorgs worth of events. It is a bit risky to lower this channel's buffer size because it might cause deadlocks.

Reply:
Why 2 reorgs?

Also, this is complex enough to warrant adding a comment here.

Reply:
Why 2 reorgs? => same question. @omerfirmak

Reply:
Oh, now that I take another look at it; I was wrong. Buffer size of 1 is enough. We ignore CCC errors from blocks that has an ancestor that fails CCC.

Reply:
what about adding some comments on the invariance guarantee here? it will help debug if the implementation breaks the assumption.

Reply:
btw, does the conclusion based on this commit: [make double-reorgs impossible](https://github.com/scroll-tech/go-ethereum/pull/982/commits/3ea2c59a736b33cbc3ec9a81444ac97ba7ed5349)?

Reply:
btw, I still cannot think of a case of the `reorgCh` deadlock here. Could you provide one?

Reply:
The main one was this: (which is no longer the case)

- Multiple blocks fail CCC and we don't ignore later ones like we do right now.
- First one gets written to the channel, second write operation blocks
- mainLoop is blocked while calling Check() so it can't read from the reorg channel
- For Check() to unblock a checker task needs to become available.
- For a checker task to become available, they need to be able to write to reorg channel 
- But they can't write to reorg channel, because mainLoop can't read from reorg channel to free up some space. Since it is blocked on Check()